x-pack/legacy/plugins/fleet/public/pages/agent_details/components/modal_confirm_unenroll.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { SFC } from 'react';
+import { EuiConfirmModal, EuiOverlayMask } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+interface Props {
+  onConfirm: () => void;
+  onCancel: () => void;
+}
+
+export const ModalConfirmUnenroll: SFC<Props> = ({ onConfirm, onCancel }) => {
```

Comment:
this component could be written as a provider pattern that could be used from any UI, for any number of agents (useful as we will need unenroll in the agent list view, at least!).

an example "delete policy(ies)" modal used in Snapshot and Restore: [policy_delete_provider.tsx](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/snapshot_restore/public/app/components/policy_delete_provider.tsx). note that it handles modal open state, making request for deletion, and displaying toast notification for failure/success  within the component. then it can be used for [multiple delete](https://github.com/elastic/kibana/blob/371a113cc9259fa8198861f80c5ae8296aaf272c/x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/policy_list/policy_table/policy_table.tsx#L301) and [single delete](https://github.com/elastic/kibana/blob/371a113cc9259fa8198861f80c5ae8296aaf272c/x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/policy_list/policy_table/policy_table.tsx#L247) anywhere!

what do you think of this pattern? just want to see if you like or dislike the pattern. no need to change in this PR unless you want to 🙂

Reply:
I am not really familiar with this pattern, maybe something we can refactor while implementing unenrolling in the listing,